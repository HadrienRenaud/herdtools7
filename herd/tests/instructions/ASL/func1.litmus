ASL func1

{}

func f(i:integer) => integer
begin
    return i;
end;

func main() => integer
begin
    let x = 3;
    let y = f(x);

    return 0;
end;

forall (
    0: main.0.x = 3 /\
    0: main.0.y = 3
)

