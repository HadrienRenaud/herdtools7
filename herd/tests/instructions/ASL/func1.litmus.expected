Test func1 Required
States 1
0:main.0.x=3; 0:main.0.y=3;
Ok
Witnesses
Positive: 1 Negative: 0
Condition forall (0:main.0.x=3 /\ 0:main.0.y=3)
Observation func1 Always 1 0
Hash=d9f68b63dd785d263d825c3b7f1144b0

